wpe/src/main/java/com/wpe/wpe/PageSettings.java

```diff
@@ -0,0 +1,89 @@
+package com.wpe.wpe;
+
+import android.annotation.SuppressLint;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class PageSettings
+{
+    private static final int DefaultMajorVersion = 12;
+    private static final int DefaultMinorVersion = 0;
+    private static final int DefaultBugfixVersion = 0;
+    private static String OS_VERSION;
+
+    @SuppressLint("DefaultLocale")
+    private static String getOsVersion()
+    {
+        if (OS_VERSION == null) {
+            String releaseVersion = android.os.Build.VERSION.RELEASE;
+            boolean validReleaseVersion = false;
+            try {
+                Pattern versionPattern = Pattern.compile("^(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)");
+                Matcher matcher = versionPattern.matcher(releaseVersion);
+                // Check if major version is found
+                String majorVersionString = matcher.group(1);
+                if (majorVersionString != null && !majorVersionString.isEmpty()) {
+                    validReleaseVersion = true;
+                }
+            } catch (Exception ignored) {}
+
+            if (validReleaseVersion) {
+                OS_VERSION = releaseVersion;
+            } else {
+                OS_VERSION = String.format("%d.%d.%d",
+                    DefaultMajorVersion,
+                    DefaultMinorVersion,
+                    DefaultBugfixVersion);
+            }
+        }
+        return OS_VERSION;
+    }
+
+    private static final String DEFAUlT_USER_AGENT =  String.format("Mozilla/5.0 (Linux; Android %s) ", getOsVersion()) +
+        "AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.0 Mobile Safari/605.1.15";
```

Comment:
As we've got a `String.format()` it would be more efficient to put everything in the first string and not using a `+` concatenation in the middle.